### PR TITLE
Use pytest < 5 to support testing on Python 2.7 and 3.4.

### PR DESCRIPTION
**Pull request by gordonmessmer**
https://github.com/pytest-dev/pytest/issues/5519

### CODE REVIEW

**Comment by hroncok** [comment]
pip should pull in the correct version on those platforms, doesn't it?

**Comment by gordonmessmer** [comment]
pip does, yes, but python2 pip doesn't seem to be used (or installed) in the test env.  

For pytest requirement with no version specified:

```
[root@433a1ce3e52c pyp2rpm]# python2.7 setup.py test --addopts -vv
...
running pytest
Searching for pytest
Reading https://pypi.org/simple/pytest/
Downloading https://files.pythonhosted.org/packages/60/23/de5604e58f4eba7a90f70486c8d4ece25f1a404bae29683903ffd2aea425/pytest-5.0.1.tar.gz#sha256=6ef6d06de77ce2961156013e9dff62f1b2688aa04d0dc244299fe7d67e09370d
Best match: pytest 5.0.1
Processing pytest-5.0.1.tar.gz
Writing /tmp/easy_install-mBQzlt/pytest-5.0.1/setup.cfg
Running pytest-5.0.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-mBQzlt/pytest-5.0.1/egg-dist-tmp-U0lQjp
  File "build/bdist.linux-x86_64/egg/_pytest/pytester.py", line 772
    def inline_run(self, *args, plugins=(), no_reraise_ctrlc=False):
                                      ^
SyntaxError: invalid syntax

```

**Comment by hroncok** [comment]
right, setup.py test_requirements do not work like this :(

File: setup.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
                     'click',
                     'Jinja2',
                     ],
-    tests_require=['pytest'],
+    tests_require=['pytest < 5'],
```
Comment (hroncok) [inline setup.py]:
```suggestion
    tests_require=['pytest < 5;python_version<"3.5"', 'pytest'],
```

This migth work.

Comment (gordonmessmer) [reply setup.py]: With "pytest < 5":

```
# python2.7 setup.py test --addopts -vv
...
running pytest
Searching for pytest<5
Reading https://pypi.org/simple/pytest/
Downloading https://files.pythonhosted.org/packages/8c/6f/4db6124788e9b281d94693e21e5a40d00f417763dcda10edc8a70655cb92/pytest-4.6.4-py2.py3-none-any.whl#sha256=6aa9bc2f6f6504d7949e9df2a756739ca06e58ffda19b5e53c725f7b03fb4aae
Best match: pytest 4.6.4
Processing pytest-4.6.4-py2.py3-none-any.whl
Installing pytest-4.6.4-py2.py3-none-any.whl to /home/gordon/git/pyp2rpm/.eggs
writing requirements to /home/gordon/git/pyp2rpm/.eggs/pytest-4.6.4-py2.7.egg/EGG-INFO/requires.txt

```

With ['pytest < 5;python_version<"3.5"', 'pytest']:

```
# python2.7 setup.py test --addopts -vv
...
running pytest
Searching for pytest
Reading https://pypi.org/simple/pytest/
Downloading https://files.pythonhosted.org/packages/60/23/de5604e58f4eba7a90f70486c8d4ece25f1a404bae29683903ffd2aea425/pytest-5.0.1.tar.gz#sha256=6ef6d06de77ce2961156013e9dff62f1b2688aa04d0dc244299fe7d67e09370d
Best match: pytest 5.0.1
...
pkg_resources.VersionConflict: (pytest 5.0.1 (/home/gordon/git/pyp2rpm/.eggs/pytest-5.0.1-py2.7.egg), Requirement.parse('pytest<5; python_version < "3.5"'))
```

Comment (gordonmessmer) [reply setup.py]: Oh, but adding a platform dependency to the second list item fixes it.  Thanks for the tip.

**Pull request merged**